prompts/aks-mcp-tool-consolidation.md

```diff
@@ -0,0 +1,238 @@
+
+## Current State Analysis
+
+### Tool Categories and Current Count
+
+1. **Azure CLI Commands** (~25 tools)
+   - AKS Commands: 14 tools (show, list, create, delete, scale, update, etc.)
+   - Monitor Commands: 3 tools (metrics operations)
+   - Account Commands: 3 tools (login, list, set)
+   - Special Commands: 5 tools (fleet, resource health, app insights, etc.)
+
+2. **Azure Resource Tools** (~10 tools)
+   - Network Tools: 6 tools (VNet, NSG, RouteTable, Subnet, LoadBalancer, PrivateEndpoint)
+   - Detector Tools: 3 tools (list, run, run by category)
+   - Compute Tools: 1+ tools (VMSS operations)
+
+3. **Control Plane Tools** (2 tools)
+   - Diagnostic settings
+   - Logs querying
+
+4. **Kubernetes Tools** (3+ tools)
+   - kubectl commands
+   - helm (optional)
+   - cilium (optional)
+
+5. **Advisor Tools** (1 tool)
+   - Advisor recommendations
+
+### Current Registration Patterns
+
+1. **Command-based tools**: Use generic "args" parameter with command executors
+2. **Resource-based tools**: Use specific parameters (subscription_id, resource_group, etc.)
+3. **Access control**: Enforced at registration time based on `cfg.AccessLevel`
+
+## Proposed Consolidated Tool Structure
+
+### 1. `az_aks_operations` - Unified AKS Management Tool
+
+**Combines:**
+- All AKS show/list/get operations
+- All AKS create/update/delete operations
+- All nodepool operations
+- Account management commands
+
+**Parameters:**
+```json
+{
+  "operation": "show|list|create|delete|scale|update|upgrade|nodepool-*|account-*",
+  "resource_type": "cluster|nodepool|account",
+  "args": "additional arguments as needed"
+}
+```
+
+**Benefits:**
+- Reduces ~20 AKS-related tools to 1
+- Maintains access control through operation validation
+- Cleaner interface for users
+
+### 2. `az_network_resources` - Unified Network Resource Query Tool
+
+**Combines:**
+- VNet info
+- NSG info
+- RouteTable info
+- Subnet info
+- LoadBalancer info
+- PrivateEndpoint info
+
+**Parameters:**
+```json
+{
+  "resource_type": "all|vnet|nsg|route_table|subnet|load_balancer|private_endpoint",
+  "subscription_id": "subscription ID",
+  "resource_group": "resource group name",
+  "cluster_name": "AKS cluster name",
+  "filters": "optional filters"
+}
+```
+
+**Benefits:**
+- Reduces 6 network tools to 1
+- Can fetch all network resources or specific types
+- More efficient for comprehensive network diagnostics
+
+### 3. `az_monitoring` - Unified Monitoring and Diagnostics Tool
+
+**Combines:**
+- Metrics operations
+- Resource health monitoring
+- App Insights queries
+- Control plane diagnostics
+- Control plane logs
+
+**Parameters:**
+```json
+{
+  "operation": "metrics|resource_health|app_insights|diagnostics|logs",
+  "query_type": "specific to operation",
+  "parameters": "operation-specific parameters"
```

Comment:
az cli support '--help', while others don't. do we have a way to let AI know it can call '--help' for some specific operation to get the guidance?

Reply:
could we make --help as part of parameters